csvcubed/csvcubed/models/csvcubedexception.py

```diff
@@ -0,0 +1,281 @@
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return CsvcubedExceptionUrls.InvalidNumberOfRecords.value
+
+
+class FailedToReadSparqlQueryException(Exception):
+    """Class representing the FailedToReadSparqlQueryException model."""
+
+    def __init__(self, sparql_file_path: Path):
+        super().__init__(
+            f"{CsvcubedExceptionMsges.FailedToReadSparqlQuery.value}: {self.get_error_url()}".format(
+                sparql_file_path=str(sparql_file_path),
+            )
+        )
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return CsvcubedExceptionUrls.FailedToReadSparqlQuery.value
+
+
+class InvalidCsvFilePathException(Exception):
+    """Class representing the InvalidCsvFilePathException model."""
+
+    def __init__(self):
+        super().__init__(
+            f"{CsvcubedExceptionMsges.InvalidCSVFilePath.value}: {self.get_error_url()}"
+        )
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return CsvcubedExceptionUrls.InvalidCSVFilePath.value
+
+
+class FailedToLoadTableSchemaIntoRDFGraphException(Exception):
+    """Class representing the FailedToLoadTableSchemaIntoRDFGraphException model."""
+
+    def __init__(self, table_schema_file: str):
+        super().__init__(
+            f"{CsvcubedExceptionMsges.FailedToLoadTableSchemaIntoRDFGraph.value}: {self.get_error_url()}".format(
+                table_schema_file=table_schema_file
+            )
+        )
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return CsvcubedExceptionUrls.FailedToLoadTableSchemaIntoRDFGraph.value
+
+
+class FailedToParseJSONldtoRDFGraphException(Exception):
+    """Class representing the FailedToParseJSONldtoRDFGraphException model."""
+
+    def __init__(self, csvw_metadata_file_path: Path):
+        super().__init__(
+            f"{CsvcubedExceptionMsges.FailedToParseJSONldtoRDFGraph.value}: {self.get_error_url()}".format(
+                csvw_metadata_file_path=str(csvw_metadata_file_path)
+            )
+        )
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return CsvcubedExceptionUrls.FailedToParseJSONldtoRDFGraph.value
+
+
+class UnsupportedComponentPropertyTypeException(Exception):
+    """Class representing the UnsupportedComponentPropertyType model."""
+
+    def __init__(self, property_type: str):
+        super().__init__(
+            f"{CsvcubedExceptionMsges.UnsupportedComponentPropertyType.value}: {self.get_error_url()}".format(
+                property_type=property_type
+            )
+        )
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return CsvcubedExceptionUrls.UnsupportedComponentPropertyType.value
+
+
+class FailedToConvertDataFrameToStringException(Exception):
+    """Class representing the FailedToConvertDataFrameToStringException model."""
+
+    def __init__(self):
+        super().__init__(
+            f"{CsvcubedExceptionMsges.FailedToConvertDataFrameToString.value}: {self.get_error_url()}"
```

Comment:
I'd rather we keep the error URL out of the message if we could. That way we can have the code which catches this exception decide how the URL should be built. I'm imaginging in the future when we have a web interface we can present the message and turn it into a hyperlink pointing to the error URL.

Reply:
If you want to change the higest level try/except so that it does something like the following then that would be great:

```python
except Exception as e:
    log_exception(_logger, e)
    if isinstance(e, HasErrorUrl):
        _logger.error(f"More information available at {e.get_error_url()}")
```

Reply:
This is now implemented.